chapter07/mdw/Ex04.pl

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use 5.010;
+
+foreach (<>) {
+  if ( /\b[^\s\\]*[A-Z]+[^\s\\]*[a-z]+[^\s\\]*\b/ || /\b[^\s\\]*[a-z]+[^\s\\]*[A-Z]+[^\s\\]*\b/ ) {
+ # comment as i have made this more complicated than needed, this will match any word with anything+capital+anything+non-capital+anything or the othe way around
+    print $_
+  }
```

Comment:
This doesn't match IBM because that is all capitals but it does match Sørina, Mátyás and Mónica.